Dockerfile

```diff
@@ -1,21 +1,28 @@
-FROM python:alpine
+FROM ubuntu:bionic
 
 WORKDIR /home/nmos-testing
 ADD . .
 
-RUN apk update \
- && apk add bash gcc musl-dev linux-headers git libffi-dev openssl-dev procps drill git coreutils libidn nodejs npm \
- && rm -rf /var/cache/apk/* \
- && pip3 install -r requirements.txt \
- && mkdir -p /config \
- && mv nmostesting/Config.py /config/Config.py \
- && ln -s /config/Config.py nmostesting/Config.py \
- && cd testssl \
- && wget https://github.com/drwetter/testssl.sh/archive/3.0rc5.tar.gz \
- && tar -xvzf 3.0rc5.tar.gz --strip-components=1 \
- && rm 3.0rc5.tar.gz \
- && npm config set unsafe-perm true \
- && npm install -g AMWA-TV/sdpoker#v0.2.0
+RUN apt-get update \
+    && export DEBIAN_FRONTEND=noninteractive \
+    && apt-get install -y --no-install-recommends \
+    gcc openssl libssl1.0-dev wget ca-certificates avahi-daemon avahi-utils libnss-mdns libavahi-compat-libdnssd-dev \
```

Comment:
Was there any reason for using libssl1.0-dev rather than libssl-dev (which pulls in v1.1)?

Reply:
Yes... but possibly libssl-dev will now work instead. IIRC when I was building up the dependencies one of the Python requirements.txt dependencies needed libssl1.0-dev rather the more standard libssl-dev. Possibly now the Dockerfile is more complete it might work with libssl-dev instead.

Reply:
Looks like it might be 'nodejs-dev' causing that requirement, which is ultimately required by npm for some reason, without which sdpoker won't install. Not ideal, but doesn't seem immediately avoidable.

Reply:
My only concern here is whether that will allow support for TLS v1.3, but I'm looking into it.

Reply:
Hmmm... It probably won't support TLS v1.3 without OpenSSL v1.1 support - Have you tried adding both libssl1.0-dev and libssl-dev? It may simply be that's its been broken up in the Ubuntu packages rather than combined in the Alpine one.

Reply:
It seems to be a known issue that won't be fixed: https://bugs.launchpad.net/ubuntu/+source/nodejs/+bug/1794589

I think we're either going to need to install NodeJS from a PPA, or perhaps easier use a more recent Ubuntu LTS:

```
diff --git a/Dockerfile b/Dockerfile
index 22cb321..5fe22a2 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -1,4 +1,4 @@
-FROM ubuntu:bionic
+FROM ubuntu:focal
 
 WORKDIR /home/nmos-testing
 ADD . .
@@ -6,8 +6,8 @@ ADD . .
 RUN apt-get update \
     && export DEBIAN_FRONTEND=noninteractive \
     && apt-get install -y --no-install-recommends \
-    gcc openssl libssl1.0-dev wget ca-certificates avahi-daemon avahi-utils libnss-mdns libavahi-compat-libdnssd-dev \
-    python3 python3-pip nodejs nodejs-dev node-gyp npm \
+    gcc openssl libssl-dev wget ca-certificates avahi-daemon avahi-utils libnss-mdns libavahi-compat-libdnssd-dev \
+    python3 python3-pip python3-dev nodejs npm \
     bash procps ldnsutils libidn11 git coreutils curl bsdmainutils \
     && rm -rf /var/lib/apt/lists/* \
     && apt-get clean -y --no-install-recommends \
```

Reply:
No - That doesn't work its mutually exclusive...